test/suite/test_util18.py

```diff
@@ -0,0 +1,128 @@
+            key = 'KEY' + str(i)
+            val = 'VAL' + str(i)
+            # Check if the KEY/VAL commits exist in the log file
+            if expect_keyval:
+                self.check_file_contains(log_file, '"key": "%s\\u0000"' % key)
+                self.check_file_contains(log_file, '"value": "%s\\u0000"' % val)
+            else:
+                self.check_file_not_contains(log_file, '"key": "%s\\u0000"' % key)
+                self.check_file_not_contains(log_file, '"value": "%s\\u0000"' % val)
+
+            # Convert our KEY/VAL strings to their expected hex value
+            hex_key = codecs.encode(key.encode(), 'hex')
+            val_key = codecs.encode(val.encode(), 'hex')
+            # Check if the KEY/VAL commits exist in the log file (in hex form)
+            if expect_keyval_hex:
+                self.check_file_contains(log_file, '"key-hex": "%s00"' % str(hex_key, 'ascii'))
+                self.check_file_contains(log_file, '"value-hex": "%s00"' % str(val_key, 'ascii'))
+            else:
+                self.check_file_not_contains(log_file, '"key-hex": "%s00"' % str(hex_key, 'ascii'))
+                self.check_file_not_contains(log_file, '"value-hex": "%s00"' % str(val_key, 'ascii'))
+
+    def test_printlog_file(self):
+        """
+        Test printlog on a populated table
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog"], outfilename='printlog.out')
+        self.check_non_empty_file('printlog.out')
+        self.check_populated_printlog('printlog.out', True, False)
+
+    def test_printlog_hex_file(self):
+        """
+        Test printlog with hexadecimal formatting on a populated table
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog", "-x"], outfilename='printlog-hex.out')
+        self.check_non_empty_file('printlog-hex.out')
+        self.check_populated_printlog('printlog-hex.out', True, True)
+
+    def test_printlog_message(self):
+        """
+        Test printlog with messages-only formatting on a populated table
+        """
+        # Run reconfigure with compatibility to force 'COMPATIBILITY' message
+        self.conn.reconfigure('compatibility=(release=3.2.0)')
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
+        self.runWt(["printlog", "-m"], outfilename='printlog-message.out')
+        self.check_non_empty_file('printlog-message.out')
+        self.check_file_contains('printlog-message.out', 'COMPATIBILITY: Version now 3')
+        self.check_populated_printlog('printlog-message.out', False, False)
+
+    def test_printlog_lsn_offset(self):
+        """
+        Test printlog with an LSN offset provided
+        """
+        self.session.create('table:' + self.tablename, self.create_params)
+        self.populate()
```

Comment:
Can you give a comment about what you're testing here and what you're expecting? It might be interesting to get the first and second LSNs (perhaps via log cursor) and then generalize it instead of hard-coding in the exact offsets. Units tests often have specific hard-coded things in them, so think about it. It may or may not be easy to do.

Reply:
That's a great point, I'm all for generalizing! I've updated the test to extract the LSN's from a log cursor.
Additionally updated the comments.